packages/transport/rn-hid/android/src/main/kotlin/com/ledger/devicesdk/shared/internal/connection/InternalConnectedDevice.kt

```
@@ -9,5 +9,5 @@ internal data class InternalConnectedDevice(
     val name: String,
     val ledgerDevice: LedgerDevice,
     val connectivity: ConnectivityType,
-    val sendApduFn: suspend (ByteArray) -> SendApduResult,
+    val sendApduFn: suspend (apdu: ByteArray, triggersDisconnection: Boolean, abortTimeoutDuration: kotlin.time.Duration) -> SendApduResult,
```

(ASK] To be honest I don't understand why do we need `triggersDisconnection` and `abortTimeoutDuration` 😅

---

- `triggersDisconnection`: in the DMK TS, we can pass this parameter to `sendApdu` if we know that a given command will trigger a disconnection if the command is successful. This allows the state machine to anticipate the disconnection.
- `abortTimeoutDuration` is needed in particular for the sync onboarding. In that onboarding, in LL, we have a polling with sends a `getOSVersion` command periodically. There is a issue with that polling right after you validate your seed phrase on the device, because the device becomes unresponsive for a few seconds: any APDU sent at this moment will just never get a reply. In order to not block the next APDUs being sent, we add this timeout and return a specific error. This error will be ignored in the sync onboarding. Without the timeout, the device session would be in "busy" state and would not allow any new APDU to be sent.

---

Ok got it. Thks